external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaSourceProviderSuite.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.kafka010
+
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import org.scalatest.PrivateMethodTester
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.sources.v2.reader.Scan
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class KafkaSourceProviderSuite extends SparkFunSuite with PrivateMethodTester {
+  private val pollTimeoutMsMethod = PrivateMethod[Long]('pollTimeoutMs)
+
+  test("SPARK-28142 - continuous mode - options should be handled as case-insensitive") {
```

Comment:
Let's reduce the repetition in this test function and remove `buildCaseInsensitiveStringMap`, too.
```
  test("SPARK-28142 - continuous mode - options should be handled case-insensitively") {
    // we're trying to read the value of "pollTimeout" to see whether option is handled correctly
    val option = KafkaSourceProvider.CONSUMER_POLL_TIMEOUT
    val expectedValue = 1000
    Seq(option.toUpperCase(Locale.ROOT), option.toLowerCase(Locale.ROOT)).foreach { name =>
      val map = new CaseInsensitiveStringMap(Map(
        "kafka.bootstrap.servers" -> "dummy",
        "subscribe" -> "dummy",
        name -> expectedValue.toString).asJava)
      assert(expectedValue === getFieldFromContinuousStream(map, pollTimeoutMsMethod))
    }
  }
```

Reply:
Same here. I guess I would just add test here to ensure they are being reused. Maybe the test would fail given #24967 is a patch for that, but we can comment here and in #24967 we can uncomment. Let me update the PR.

Reply:
? What is the same here? At least you should remove the repeatition.

Reply:
I'm sorry I missed to get the point. Looks like the point is reducing code for testing uppercase and lowercase. Thanks for the great suggestion. Addressed.